widget/tree.go

```diff
@@ -186,6 +188,40 @@ func (t *Tree) Selection() string {
 // SetSelection updates the current selection to the node with the given Unique ID.
 func (t *Tree) SetSelection(uid string) {
 	t.selected = uid
+	if t.selected != TreeNoSelection && t.scroller != nil {
+		var found bool
+		var y int
+		var size fyne.Size
+		t.walkAll(func(uid string, branch bool, depth int) {
+			m := t.leafMinSize
+			if branch {
+				m = t.branchMinSize
+			}
+			if uid == t.selected {
+				found = true
+				size = m
+			} else if !found {
```

Comment:
Can we not return early when uid == t.selected and found is then set true  thus not needing the else if and removing the indention?

Reply:
Once we start walking the tree we can't abort once we find the node. The code would become longer;
```suggestion
				return
			}
			if !found {
```

Reply:
You wouldn't need the !found would you because if we made it this far and we returned early then we know the node wasn't found

Reply:
It would be possible to make the callback return bool so walk could determine whether to continue, though this PR would get larger

Reply:
I was missing the part where found had to be persisted.  Gotcha.